modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/NettyTcpChannel.java

```diff
@@ -98,8 +98,11 @@ public void sendMessage(BytesReference reference, ActionListener<Void> listener)
             } else {
                 final Throwable cause = f.cause();
                 Netty4Utils.maybeDie(cause);
-                assert cause instanceof Exception;
-                listener.onFailure((Exception) cause);
+                if (cause instanceof Error) {
```

Comment:
None. Elasticsearch should still die on another thread. We just have to do something on this thread in the meantime. In the old version the assertion would only trigger is assertions were enabled. And then it would throw a class cast exception. 

I mean this is super edge case, but I thought that elasticsearch dying on another thread would handle the Error and on this thread we would just continue on in the meantime.